indico/core/db/sqlalchemy/descriptions.py

```diff
@@ -20,14 +20,32 @@
 from sqlalchemy.ext.hybrid import hybrid_property
 
 from indico.core.db import db
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.util.string import MarkdownText, RichMarkup
+from indico.util.struct.enum import TitledIntEnum
+
+
+class RenderMode(TitledIntEnum):
+    """Rendering formats that a description can be written in."""
+
+    __titles__ = [None, 'HTML', 'Markdown', 'Plain Text']
+    html = 1
+    markdown = 2
+    plain_text = 3
+
+
+RENDER_MODE_WRAPPER_MAP = {
+    RenderMode.html: RichMarkup,
+    RenderMode.markdown: MarkdownText,
+    RenderMode.plain_text: unicode
```

Comment:
I don't think `unicode` works here. You need a custom `Markup` subclass that wraps it in `div.preformatted` (you can probably copy `RichMarkup` and remove the guessing code)